### DISCUSSION TITLE
building Airlfow using custom image Using RHEL/Centos

### DISCUSSION THREAD

**Discussion by arunpatcha** [apache/airflow#19187] [Q&A]
Hi 

 I am trying to build Aiflow from source using RHEL/CENTOS image i am getting issues with default python version after building image . the python is defaulting to python 2.7 . have tried multiple changes. 
 1. setting python to python3 (using airlfow source code is breaking after we perform container deployment)
 2. creating base image for python3 and using it ( its breaking yum as its requires python2) 

any help here would be great :)

**Comment by potiuk** [comment]
What is the reason you want to use RHEL/CENTOS? 

We are using debian and I recommend using it instead (especially latest versions that do not even have python 2.7 installed any more). 

If you do want to us even older ubuntu/debian distros you can install "python-is-python3" package to get python by default https://askubuntu.com/questions/320996/how-to-make-python-program-command-execute-python-3 

If for any  reason  you want to use  RHEL /centos  you  have some ways https://stackoverflow.com/questions/45542690/how-to-set-python3-5-2-as-default-python-version-on-centos


And finally  - there  there is a long and heated  discussion at  PIP Github where the conclusion is that even in images  you should use virtualenv to install anythoing rather than rely  on default python installation . I am actually working on converting the officiial  (debiaan baad ) image  to  use virtualenv.  - https://github.com/pypa/pip/issues/10556

**Reply by arunpatchaivy** [reply]
Hi ,

Sorry for delayed response. 
we have a security requirement  they have been pushing us to use oraclelinux/RHEL  as base images.

I agree with you regarding use of Debian images usage in Airflow.  

we done some modification  so that yum does not break after setting python3 as default

Update : 
we were able to successfully build the custom  image using airflow source . But there are lot unknown for us currently we are checking if dag run works .  currently we are using sample dag provided by airflow 
https://airflow.apache.org/docs/apache-airflow/stable/_modules/airflow/example_dags/tutorial.html

we are getting something like this .


*** Unable to fetch logs from worker pod tutorialprintdate.ddcdf3efd5e34b288f89a6ec1227e0c8 ***
(404)
Reason: Not Found
HTTP response headers: HTTPHeaderDict({'Audit-Id': '807c1d34-c1e8-4f75-9140-18c7de8f471e', 'Cache-Control': 'no-cache, private', 'Content-Type': 'application/json', 'Date': 'Mon, 01 Nov 2021 04:58:47 GMT', 'Content-Length': '268'})
HTTP response body: b'{"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"pods \\"tutorialprintdate.ddcdf3efd5e34b288f89a6ec1227e0c8\\" not found","reason":"NotFound","details":{"name":"tutorialprintdate.ddcdf3efd5e34b288f89a6ec1227e0c8","kind":"pods"},"code":404}\n'
